robot/impl/local_robot_test.go

```diff
@@ -3482,3 +3482,26 @@ func TestCustomResourceBuildsOnModuleAddition(t *testing.T) {
 	)
 	rtestutils.VerifySameResourceNames(t, r.ResourceNames(), expectedResources)
 }
+
+func TestSendTriggerConfig(t *testing.T) {
+	// Tests that the triggerConfig channel buffers exactly 1 message and that
+	// sendTriggerConfig does not block whether or not the channel has
+	// capacity.
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	// Set up local robot normally so that the triggerConfig channel is set up normally
+	r := setupLocalRobot(t, ctx, &config.Config{}, logger)
+
+	// Close the robot to stop the background workers from processing any messages to triggerConfig
+	// but also reinitialize the closeContext so that sendTriggerConfig will attempt to send messages
+	// through.
+	r.Close(ctx)
```

Comment:
[minor] we expect to this to not error right?

```suggestion
	test.That(t, r.Close(ctx), test.ShouldBeNil)
```

Reply:
or more importantly, would there be an issue if there were an error?

Reply:
I suppose so, that may mean that the closeContext did not get closed properly